api/src/main/java/jakarta/data/repository/Streamable.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.repository;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Simple interface to ease streamability of {@link Iterable}s.
+ * This is interface and can therefore be used as the assignment target for a lambda expression or method reference.
+ */
+@FunctionalInterface
+public interface Streamable<T> extends Iterable<T>, Supplier<Stream<T>> {
+
+    /**
+     * Creates a non-parallel {@link Stream} of the underlying {@link Iterable}.
+     *
+     * @return a non-parallel Stream
+     */
+    default Stream<T> stream() {
```

Comment:
I wouldn't actually do that, as the presence of `get` is a necessary side effect of implementing `Supplier`. The `stream()` method is much more explicit and also in alignment with `Collection.stream()`, which `Streamable` is usually more of a replacement for. If in doubt, I'd even rather go ahead and remove the `Supplier` implementation. It was merely added in Spring Data to be able to hand a `Streamable` into methods that expect a `Supplier<Stream>` but that's definitely not the primary use case of it.

Reply:
I'd be fine with keeping `stream()` and getting rid of `Supplier` / `get()`.  My intent was just to avoid the redundancy and what you've suggested will achieve that.

Reply:
yeah removing `Supplier<Stream>` makes sense to me

Reply:
Done.